### PR TITLE
Add ci github workflow for pytest

### PR DESCRIPTION
**Pull request by IvanObreshkov**

### CODE REVIEW

File: .github/workflows/pytests.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
```
Comment (asynchroza) [inline .github/workflows/pytests.yml:32]: Should probably cache those

Comment (IvanObreshkov) [reply .github/workflows/pytests.yml]: @asynchroza Fixed in 29afd142bb53998e72a12beac782052f46646bcf

**Comment by IvanObreshkov** [comment]
> @asynchroza We should be good to go: https://github.com/AUBGTheHUB/monolith/actions/runs/10884168999 You can also download the cov report in html format.

What is this guy talking about :D

File: .github/workflows/pytests.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      - name: Install Poetry And Dependiecies ${{ matrix.python-version }}
+        run: pipx install poetry
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'poetry'
+      - run: poetry install
+
+      - name: Run tests
+        run: poetry run pytest --cov --cov-report=html
```
Comment (asynchroza) [inline .github/workflows/pytests.yml:39]: What do you need the `--cov-report=html` for?

Comment (asynchroza) [reply .github/workflows/pytests.yml:39]: Nvm, just saw that you're uploading the coverage.

File: .github/workflows/pytests.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+name: Unit and Integration tests with pytest
+
+on:
+  pull_request:
+    types: [ assigned, opened, synchronize, reopened ]
+    #    paths: [ 'services/py_api/**' ]
```
Comment (asynchroza) [inline .github/workflows/pytests.yml:6]: Probably uncomment and make it detect only *.py code, toml and lockfile changes

Comment (IvanObreshkov) [reply .github/workflows/pytests.yml]: @asynchroza Fixed in [9cc55f2](https://github.com/AUBGTheHUB/monolith/pull/714/commits/9cc55f22a6ea9470d7b4ccc18d592f7c9f2a6ed9)

**Review by asynchroza — APPROVED** [review]
LGTM

**Comment by asynchroza** [comment]
Just be mindful that your requirements for when to run the workflow might change in the future. If you were to use json files to load certain data somewhere, changes in these files wouldn't trigger this workflow.

**Pull request merged**